airflow/providers/google/cloud/hooks/compute.py

```diff
@@ -155,88 +503,167 @@ def _execute_set_machine_type(self, zone: str, resource_id: str, body: dict, pro
+        :param request_id: Unique request_id that you might add to achieve
+            full idempotence (for example when client call times out repeating the request
+            with the same request id will not create a new Instance Group Managers again)
+            It should be in UUID format as defined in RFC 4122
+        :param project_id: Google Cloud project ID where the Compute Engine Instance Group Managers exists.
+            If set to None or missing, the default project_id from the Google Cloud connection is used.
+        :param zone: Google Cloud zone where the Instance exists
+        :param retry: A retry object used to retry requests. If `None` is specified, requests
+            will not be retried.
+        :param timeout: The amount of time, in seconds, to wait for the request to complete.
+            Note that if `retry` is specified, the timeout applies to each individual attempt.
+        :param metadata: Additional metadata that is provided to the method.
         """
-        response = (
-            self.get_conn()
-            .instanceTemplates()
-            .get(project=project_id, instanceTemplate=resource_id)
-            .execute(num_retries=self.num_retries)
+        client = self.get_compute_instance_group_managers_client()
+        client.insert(
+            # Calling method insert() on client to create the specified Instance Group Managers.
+            # This method accepts request object as an argument and should be of type
+            # Union[google.cloud.compute_v1.types.InsertInstanceGroupManagerRequest, dict] to construct
+            # a request message.
+            # The request object should be represented using arguments:
+            #   instance_group_manager_resource (google.cloud.compute_v1.types.InstanceGroupManager):
+            #       The body resource for this request.
+            #   project (str):
+            #       Project ID for this request.
+            #   zone (str):
+            #       The name of the zone where you want to create the managed instance group.
+            #   request_id (str):
+            #       An optional request ID to identify requests.
+            request={
+                'instance_group_manager_resource': body,
+                'project': project_id,
+                'zone': zone,
+                'request_id': request_id,
+            },
+            retry=retry,
+            timeout=timeout,
+            metadata=metadata,
         )
-        return response
 
     @GoogleBaseHook.fallback_to_default_project_id
-    def insert_instance_template(
+    def get_instance_group_manager(
         self,
-        body: dict,
+        resource_id: str,
+        zone: str,
         project_id: str = PROVIDE_PROJECT_ID,
-        request_id: Optional[str] = None,
-    ) -> None:
+        retry: Optional[Retry] = None,
+        timeout: Optional[float] = None,
+        metadata: Optional[Sequence[Tuple[str, str]]] = (),
+    ) -> dict:
```

Comment:
I think Instance Group Manger Object will be better than Dict

Reply:
Done